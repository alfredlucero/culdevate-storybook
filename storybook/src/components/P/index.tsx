import * as React from 'react';

import styled, { css } from 'styled-components';

export interface PProps {
  small?: boolean;
  danger?: boolean;
  success?: boolean;
  socialColor?: 'cudos' | 'comment' | 'share';
}

const P = styled.p`
  color: #595959;
  font-size: ${(props:PProps) => props.small ? '0.6rem' : '1rem'};
  font-family: 'Roboto';
  line-height: 1.5;
  letter-spacing: 1px;

  ${(props:PProps) => props.success && css`
    color: #2bad88;
  `}

  ${(props:PProps) => props.danger && css`
    color: #fb8597;
  `}

  ${(props:PProps) => props.socialColor === 'cudos' && css`
    color: #f7baa3;

    &:hover {
      color: #cc9885;
    }
  `}

  ${(props:PProps) => props.socialColor === 'comment' && css`
    color: #a0a0a0;

    &:hover {
      color: #757575;
    }
  `}

  ${(props:PProps) => props.socialColor === 'share' && css`
    color: #83b9ff;

    &:hover {
      color: #6b98d2;
    }
  `}
`;

export default P;