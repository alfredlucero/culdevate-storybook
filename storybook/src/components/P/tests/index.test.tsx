import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import P from '../index';

describe('<H3 />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<P />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render its text', () => {
    const children = 'Sample P Text';
    const renderedComponent = shallow(
      <P>{children}</P>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });
});

