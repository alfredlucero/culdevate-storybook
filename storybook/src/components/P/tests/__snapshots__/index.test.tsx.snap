// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H3 /> should render correctly 1`] = `
.c0 {
  color: '#595959';
  font-size: 1rem;
  font-family: 'Noto Sans';
  line-height: 1.5;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

<p
  className="c0"
/>
`;
